internal/pkg/apikey/auth.go

```diff
@@ -36,21 +37,23 @@ func (k ApiKey) Authenticate(ctx context.Context, es *elasticsearch.Client) (*Se
 	res, err := req.Do(ctx, es)
 
 	if err != nil {
-		return nil, fmt.Errorf("apikey auth request %s: %w", k.Id, err)
+		return nil, fmt.Errorf("apikey auth request %s: %w", k.ID, err)
 	}
 
 	if res.Body != nil {
-		defer res.Body.Close()
+		defer func() {
+			_ = res.Body.Close()
+		}()
```

Comment:
I use this pattern to get the linter to stop (mostly when defering the Close). the other option is to use `//nolint:errcheck` comments, is there a preference?

Reply:
this might actually be me overreaching, when i run the linter in vim it's giving me more errors then through make